Expensify/App
# [WIP] Make text on native platforms non-selectable

### Details
- Extracts two new components, `ReportActionItemCommentFragment` and `ReportActionItemTextFragment`, which are the same on web/desktop and native, with one difference. Native text is _not_ selectable, while web/desktop text _is_ selectable. This fixes the bug linked in the description, without breaking native copy/paste functionality on web/desktop, where most users would use a mouse instead of the context menu to copy/paste text anyways.
- There was one other [common bug](https://stackoverflow.com/questions/49284127/react-native-two-taps-required-for-onpress-to-work) that occurred where, when the keyboard was open, the first tap would close the keyboard and the second `longPress` would open the `ReportActionContextMenu`. I was able to fix this so that the first `longPress` closes the keyboard and opens the `ReportActionContextMenu` in one go, using the `keyboardShouldPersistTaps` prop of `FlatList` (inherited from `ScrollView` I believe).

### Fixed Issues
Fixes https://github.com/Expensify/Expensify.cash/issues/2279

### Tests / QA Steps (Web/Desktop)
1. Make sure you can highlight text and copy/paste like normal.
2. Make sure that right-clicking a chat opens the `ReportActionContextMenu`.
3. Make sure that clicking anywhere outside the `ReportActionContextMenu` closes it.

### Tests / QA Steps (iOS/Android)
1. Open a report. If necessary click on the chat composer to open the keyboard.
2. Tap above the keyboard - make sure the keyboard is dismissed.
3. Open the keyboard again.
4. Swipe the keyboard down - make sure you can dismiss it by swiping.
5. Long-press a chat, make sure the `ReportActionContextMenu` opens
6. Tap outside or swipe to close the `ReportActionContextMenu`
7. Open the keyboard again.
8. Long-press a chat, verify that the keyboard closes and the `ReportActionContextMenu` opens. You _should not_ see the native platform's "copy" popup, as shown in the linked issue.
9. Copy the message.
10. Paste the message somewhere to verify that the correct message was copied to the clipboard.
11. Open the keyboard again.
12. Long-press a user's name in the chat. Verify that the keyboard closes and the `ReportActionContextMenu` opens. You _should not_ see the native platform's "copy" popup, as shown in the linked issue.

### Tests / QA Steps (mWeb)
On mobile web, this PR should cause no notable difference.

### Tested On


### Screenshots

#### Web
![image](https://user-images.githubusercontent.com/47436092/115091502-8cf02f00-9ecc-11eb-8aa2-e5726dad1725.png)

#### Mobile Web


#### Desktop
![image](https://user-images.githubusercontent.com/47436092/118160592-5924fe00-b3d3-11eb-876b-d971193fae08.png)

#### iOS


#### Android
